prime-router/src/main/kotlin/azure/GetSubmissions.kt

```diff
@@ -0,0 +1,80 @@
+package gov.cdc.prime.router.azure
+
+import com.microsoft.azure.functions.HttpMethod
+import com.microsoft.azure.functions.HttpRequestMessage
+import com.microsoft.azure.functions.HttpResponseMessage
+import com.microsoft.azure.functions.annotation.AuthorizationLevel
+import com.microsoft.azure.functions.annotation.FunctionName
+import com.microsoft.azure.functions.annotation.HttpTrigger
+import gov.cdc.prime.router.tokens.OktaAuthentication
+import org.apache.logging.log4j.kotlin.logger
+
+/**
+ * Submissions API
+ * Returns a list of Actions from `public.action`.
+ */
+
+class GetSubmissions(
+    submissionsFacade: SubmissionsFacade = SubmissionsFacade.common,
+    oktaAuthentication: OktaAuthentication = OktaAuthentication(PrincipalLevel.SYSTEM_ADMIN)
+) {
+    private val facade = submissionsFacade
+    private val oktaAuthentication = oktaAuthentication
+
+    /**
+     * An Azure Function that is triggered at the `/api/submissions/` endpoint
+     *
+     * @param qLimit a user supplied page size limit overwriting the default value.
+     * @param limit a default value for the number of results to display per page.
+     * @return a list of submission history results.
+     */
+    @FunctionName("getSubmissions")
+    fun run(
+        @HttpTrigger(
+            name = "getSubmissions",
+            methods = [HttpMethod.GET],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "submissions"
+        ) request: HttpRequestMessage<String?>,
+    ): HttpResponseMessage {
+        val qLimit = request.queryParameters.getOrDefault("limit", "10")
+
+        if (isNumber(qLimit)) {
+            val limit = qLimit.toInt()
+            return getList(request, limit)
+        }
+
+        return HttpUtilities.bad(request, "Limit must be an integer.")
+    }
+
+    /**
+     * Utility function for checking if a `String` is numeric.
+     * Should be moved to a separate Utility Class.
+     *
+     * @return `true` if a String is not empty and the characters are digits. Otherwise, return `false`.
+     */
+    private fun isNumber(s: String): Boolean {
+        return if (s.isNullOrEmpty()) false else s.all { Character.isDigit(it) }
```

Comment:
Check for a non-zero positive integer, rename function to something that makes sense (isPositiveInteger?)